CIP-param-application/README.md

```diff
@@ -0,0 +1,113 @@
+---
+CIP: ????
+Title: On-Chain Parameter Application
+Status: Proposed
+Category: Plutus
+Authors:
+    - Philip DiSarro <info@anastasialabs.com>
+    - Keyan M. <keyanmaskoot@gmail.com>
+    - Microproofs <kwhitemsg@gmail.com>
+Implementors: []
+Discussions: 
+    - https://github.com/cardano-foundation/CIPs/pull/934/
+Created: 2024-11-06
+License: CC-BY-4.0
+---
+
+## Abstract
+We propose a new Plutus builtin function `builtinApplyParams` with the
+signature `BuiltinByteString -> [BuiltinData] -> ScriptHash` in order to enable
+simple and low cost on-chain check for validating instances of unapplied
+scripts.
+
+## Motivation: why is this CIP necessary?
+Complex contracts tend to depend on multiple scripts, and in some cases,
+validation from scripts with some unknown parameters are required.
+
+As a simple example, assume a minting script which needs to validate that the
+destination of its token is a parameterized spending script that is uniquely
+instantiated for a user.
+
+The [current solution](#alternatives), is costly and relatively complex. With
+our proposed builtin function, the minting script of the example above can
+be provided with the unapplied script (either directly or indirectly), and have
+arbitrary values applied to it as parameters.
+
+## Specification
+
+### Function Definition
+
+We propose the Plutus builtin function `builtinApplyParams`, with the following
+type signature:
+```hs
+builtinApplyParams :: BuiltinByteString -> [BuiltinData] -> ScriptHash
```

Comment:
Why do you return a `ScriptHash` and not a `BuiltinByteString`?  Besides this `ScriptHash` not being a plutus type, would it not be nice to return the applied CBOR, so that one can apply parameters over multiple transactions.

Reply:
My bad :no_mouth: I confused the datatype with the one from Aiken (which is just an alias there).